### ISSUE TITLE
0.4.38 cannot list secrets

### ISSUE THREAD

**Issue by orolhawion** [Nitrokey/pynitrokey#405]
After I upgraded to `0.4.38` I did the following:

1. add password to password safe (3 entries overall)
2. tried to list secrets which did not work

(!) I do not know if 1 and 2 are related

```
$ nitropy nk3 secrets list
Command line tool to interact with Nitrokey devices 0.4.38
Please provide PIN to show PIN-protected entries (if any), or press ENTER to skip
Please touch the device if it blinks
Current PIN (8 attempts left):
Please touch the device if it blinks
Critical error:
An unhandled exception occurred
	Exception encountered: OSError('Failed reading a response')
Warning: no backend was found to use for communication. Please refer to documentation how to install additional libraries.
```
The NK3C then is red permanently and no more actions are possible for the following reason:

```
nitropy nk3 secrets list
Command line tool to interact with Nitrokey devices 0.4.38
Critical error:
An unhandled exception occurred
	Exception encountered: CtapError('CTAP error: 0x06 - CHANNEL_BUSY')
Warning: no backend was found to use for communication. Please refer to documentation how to install additional libraries.
```

**Comment by szszszsz** [comment]
Hi,

1. Did you use PIN encryption or touch confirmation on these newly added credentials?
2. Can you tell what firmware version did you update from on your Nitrokey 3? I assume it is the latest right now (please confirm).
4. Is it feasible for you (as in, do you have data there which cannot be lost) to call factory reset on the secrets app (and try then again afterwards)?

**Comment by szszszsz** [comment]
Another question: did you have any credentials set up before?

**Comment by orolhawion** [comment]
> Hi,
> 
> 
> 
> 1. Did you use PIN encryption or touch confirmation on these newly added credentials?

PIN encryption

> 2. Can you tell what firmware version did you update from on your Nitrokey 3? I assume it is the latest right now (please confirm).

I am on 1.5.0 and previously had 1.4.0.rc2

> 4. Is it feasible for you (as in, do you have data there which cannot be lost) to call factory reset on the secrets app (and try then again afterwards)?

No problem at all, I am not using nk3 in a production environment.

**Comment by szszszsz** [comment]
Alright. I plan to test this again on Friday. In any case, this should be fixable in firmware, and no user data should be lost in between. I think `secrets reset` call will fix it, in case you do not want to wait.

**Comment by szszszsz** [comment]
Right, the help screen misses this in the command's description - you can use `remove`:

> remove                  Remove OTP credential.

To be corrected.

**Comment by szszszsz** [comment]
Hey,

I could not reproduce this problem today. I have moved from 1.4.0-rc1 onwards and added successfully credentials there.
1. Can you tell which NK3 model do you have? Is it Nitrokey 3 NFC or Nitrokey 3 Mini?
2. Were you using OTPs in the earlier firmware versions as well? (perhaps I need to create them in 1.3.0, and see if this makes the problem)
3. Can you make a new credential?
4. Can you paste what `nitropy nk3 status` says? (can be with UUID omitted).

Another hipothesis is, that one of the files got corrupted and makes the listing not working anymore.

**Comment by orolhawion** [comment]
> Hey,
> 
> 
> 
> I could not reproduce this problem today. I have moved from 1.4.0-rc1 onwards and added successfully credentials there.
> 
> 1. Can you tell which NK3 model do you have? Is it Nitrokey 3 NFC or Nitrokey 3 Mini?
Nitrokey 3C NFC
> 
> 2. Were you using OTPs in the earlier firmware versions as well? (perhaps I need to create them in 1.3.0, and see if this makes the problem)

I think I tried that first with 1.4.0.rc1.

> 
> 3. Can you make a new credential?
Yes

> 
> 4. Can you paste what `nitropy nk3 status` says? (can be with UUID omitted).
Not right now, but later I'll do that.

**Comment by orolhawion** [comment]
As requested:

```
$ nitropy nk3 status
Command line tool to interact with Nitrokey devices 0.4.38
UUID:               <omitted>
Firmware version:   v1.5.0
Init status:        ok
Free blocks (int):  33
Free blocks (ext):  458
Variant:            LPC55
```
